src/myna/application/deer/mesh.py

```diff
@@ -0,0 +1,123 @@
+#
+# Copyright (c) 2024 Oak Ridge National Laboratory.
+#
+# This file is part of Myna. For details, see the top-level license
+# at https://github.com/ORNL-MDF/Myna/LICENSE.md.
+#
+# License: 3-clause BSD, see https://opensource.org/licenses/BSD-3-Clause.
+#
+"""Functionality for operating on Exodus mesh input for a Deer simulation"""
+
+import numpy as np
+
+
+class NetCDF4Dataset:
+    """Class to load and calculate NetCDF4 mesh properties relevant to the Deer app.
+
+    Assumes that the NetCDF4 file was generated by Cubit/Sculpt. See Myna's
+    cubit/vtk_to_exodus_region app output for an example.
+    """
+
+    def __init__(self, data_file):
+        """Load the mesh and calculate basic properties.
+
+        Args:
+            data_file: (str) path to NetCDF data file to load
+        """
+        try:
+            from netCDF4 import Dataset  # pylint: disable=import-outside-toplevel
+        except ImportError as exc:
+            raise ImportError(
+                "Myna deer app requires `pip install .[netcdf]` optional dependencies!"
+            ) from exc
+
+        self.data_file = data_file
+        with Dataset(self.data_file) as mesh:
+            self.set_dimensions(mesh)
+            self.set_n_blocks(mesh)
+            self.set_max_block_number(mesh)
+
+    def set_dimensions(self, mesh):
+        """Calculates the coordinate dimensions of the `netCDF4.Dataset` object
+
+        Args:
+            mesh: `netCDF4.Dataset` object
+        """
+        coordx = mesh.variables["coordx"]
+        coordy = mesh.variables["coordy"]
+        coordz = mesh.variables["coordz"]
+        self.bounds = np.array(
+            [
+                [np.min(coordx), np.max(coordx)],
+                [np.min(coordy), np.max(coordy)],
+                [np.min(coordz), np.max(coordz)],
+            ]
+        )
+        self.x_range = self.bounds[0, 1] - self.bounds[0, 0]
+        self.y_range = self.bounds[1, 1] - self.bounds[1, 0]
+        self.z_range = self.bounds[2, 1] - self.bounds[2, 0]
+
+    def set_n_blocks(self, mesh):
+        """Calculates the number of blocks in the `netCDF4.Dataset` object
+
+        Args:
+            mesh: `netCDF4.Dataset` object
+        """
+        self.n_blocks = mesh.dimensions["num_el_blk"].size
+
+    def set_max_block_number(self, mesh):
+        """Finds the largest block number in the `netCDF4.Dataset` object
+
+        Args:
+            mesh: `netCDF4.Dataset` object
+        """
+
+        self.max_block_number = np.max(mesh.variables["eb_prop1"][:].data)
+
+
+def get_mesh_dimensions(exodus_mesh_file):
```

Comment:
Seems easier to me to remove these wrapper functions and just instantiate the class where you need it